README.md

```diff
@@ -7,9 +7,57 @@
 
 Crossing the road between Haskell and Elm.
 
+## What this library about?
+
+`Elm-street` allows you to generate automatically derived from Haskell types
+definitions of Elm data types, JSON encoders and decoders. This helps to avoid
+writing and maintaining huge chunk of boilerplate code when developing full-stack
+applications.
+
+## Getting started
+
+In order to use `elm-street` features, you need to perform the following steps:
+
+1. Add `elm-street` to the dependencies of your Haskell package.
+2. Derive the `Elm` typeclass for relevant data types. This can be done like this:
+   ```haskell
+   import Elm (Elm)
+
+   data User = User
+       { userName :: Text
+       , userAge  :: Int
+       } deriving (Generic)
+         deriving anyclass (Elm)
+   ```
+   > **NOTE:** This requires extensions `-XDerivingStrategies`, `-XDeriveGeneric`, `-XDeriveAnyClass`.
+3. Create list of all types you want to expose to Elm:
+   ```haskell
+   type Types =
+      '[ User
+       , Status
+       ]
+   ```
+   > **NOTE:** This requires extension `-XDataKinds`.
+4. Use `generateElm` function to output definitions to specified directory under
+   specified module prefix.
+   ```
+   main :: IO ()
+   main = generateElm @Types $ defaultSettings "frontend/src" ["Core", "Generated"]
+   ```
+   > **NOTE:** This requires extension `-XTypeApplications`.
+
+   The above command when called generates the following files:
+
+     + `frontend/src/Core/Generated/Types.elm`: `Core.Generated.Types` module with the definitions of all types
+     + `frontend/src/Core/Generated/Encoder.elm`: `Core.Generated.Encoder` module with the JSON encoders for the types
+     + `frontend/src/Core/Generated/Decoder.elm`: `Core.Generated.Decoder` module with the JSON decoders for the types
+     + `frontend/src/Core/Generated/ElmStreet.elm`: `Core.Generated.ElmStreet` module with bundled helper functions
+
```

Comment:
We should also add the section about automatical call of `elm-street`, but this can be done later when we will figure out and use it by ourselves.